check_outputs.py

```diff
@@ -1,39 +1,113 @@
-    mns = fid['input'].attrs['mean']
-    shp.insert(0, -1)
-    # reshape to be (N, H, W, C)
-    val_imgs = np.array(fid['input']).reshape(shp).transpose((0,2,3,1))
+from segnet_neon_backend import NervanaGPU_Upsample
+from segnet_neon_backend import _get_bprop_upsampling
+from pixelwise_dataloader import PixelWiseImageLoader
+from segnet_neon import gen_model
+global _get_bprop_upsampling
 
-    # add the mean value back in for each channel
-    for ind in range(3):
-        val_imgs[:,:,:,ind] += mns[ind]
+parser = argparse.ArgumentParser(description='compare output of trained model to ground truth')
+parser.add_argument('image_path', type=str, help='path to the CamVid data set in the SegNet-Tutorial '
+                                                 'GitHub repo')
+parser.add_argument('save_model_file', type=str, help='serialized model file')
+parser.add_argument("--num_classes", type=int, default=12, help="number of annotated classes")
+parser.add_argument("--height", type=int, default=256, help="image height")
+parser.add_argument("--width", type=int, default=512, help="image width")
+args = parser.parse_args()
 
+# check that image dimensions are powers of 2
+if((args.height & (args.height - 1)) != 0):
+    raise TypeError("Height must be a power of 2.")
+if((args.width & (args.width - 1)) != 0):
+    raise TypeError("Width must be a power of 2.")
 
-import numpy as np
-import matplotlib.pyplot as plt
+(c, h, w) = (args.num_classes, args.height, args.width)
+
+# need to use the backend with the new upsampling layer implementation
+be = NervanaGPU_Upsample(device_id=0)
 
-# plot the first input image
-plt.imshow(val_imgs[0]/256)
+# set batch size
+be.bsz = 1
 
-# get the pixelwise classification from the softmax SegNet outputs
-img_segment = np.argmax(valout[0].reshape((12, 256, 512)), axis=0)
+# couple backend to global neon object
+NervanaObject.be = be
+
+shape = dict(channel_count=c, height=h, width=w, subtract_mean=False)
+test_params = ImageParams(center=True, flip=False,
+                          scale_min=min(h, w), scale_max=min(h, w),
+                          aspect_ratio=0, **shape)
+common = dict(target_size=h*w, target_conversion='read_contents',
+              onehot=False, target_dtype=np.uint8, nclasses=c)
+data_dir = args.image_path
+
+test_set = PixelWiseImageLoader(set_name='test', repo_dir=data_dir,media_params=test_params, 
+                                index_file=os.path.join(data_dir, 'test_images.csv'), **common)
+
+
+# initialize model object
+segnet_model = Model(layers=gen_model())
+segnet_model.initialize(test_set)
+
+# load up the serialized model
+model_desc = ModelDescription(load_obj(args.save_model_file))
+for layer in segnet_model.layers_to_optimize:
+    name = layer.name
+    trained_layer = model_desc.getlayer(name)
+    layer.load_weights(trained_layer)
 
 plt.figure()
-plt.imshow(img_segment)
+plt.ion()
+
+im1 = None
+im2 = None
+
+for x, t in test_set:
```

Comment:
Can we have this save the plots in image format instead of display? Customers probably wont' know how to pipe the display output from terminal onto their screen.